subworkflows/local/amr_annotation.nf

```diff
@@ -0,0 +1,103 @@
+// Importing the necessary modules for the workflow
+
+include { ABRITAMR_RUN } from '../../modules/nf-core/abritamr/run/main'
+include { RGI_MAIN } from '../../modules/nf-core/rgi/main/main'
+include { AMRFINDERPLUS_UPDATE } from '../../modules/nf-core/amrfinderplus/update/main'
+include { AMRFINDERPLUS_RUN } from '../../modules/nf-core/amrfinderplus/run/main'
+include { ABRICATE_RUN } from '../../modules/nf-core/abricate/run/main'
+include { ABRICATE_SUMMARY } from '../../modules/nf-core/abricate/summary/main'
+
+// This will be implemented in the future when the database
+// information is ready for hamronization tool to parse
+
+/*
+include { HAMRONIZATION_SUMMARIZE } from '../../modules/nf-core/hamronization/summarize/main'
+include { HAMRONIZATION_AMRFINDERPLUS } from '../../modules/nf-core/hamronization/amrfinderplus/main'
+include { HAMRONIZATION_RGI } from '../../modules/nf-core/hamronization/rgi/main'
+include { HAMRONIZATION_ABRICATE } from '../../modules/nf-core/hamronization/abricate/main'
+*/
+
+// Defining the workflow
+workflow AMR_ANNOTATION{
+    // Defining the input channel
+    take:
+        ch_genome
+
+    // Defining the main process
+    main:
+        // Initializing empty channels for the output files
+        ch_versions = Channel.empty()
+        abricate_report = Channel.empty()
+        abricate_summary_report = Channel.empty()
+        rgi_report = Channel.empty()
+        amrfinderplus_report = Channel.empty()
+        abritamr_report = Channel.empty()
+
+
+        // Running abricate
+        if (!params.skip_abricate){
+            abricate_dbs = Channel.empty()
+            if (params.abricate_db != null){
+                abricate_dbs = Channel.from(params.abricate_db.tokenize(","))
+            }
+            else{
+                abricate_dbs = []
+            }
+            ABRICATE_RUN(ch_genome, abricate_dbs)
+            abricate_report = ABRICATE_RUN.out.report
+            ch_versions = ABRICATE_RUN.out.versions
+            if (!params.skip_abricate_summary){
+                reports = abricate_report.collect()
```

Comment:
I would generate a separate report for each database to distinguish between the database origins of the gene hits.

Reply:
@jimmyliu1326 I feel this is a conceptual decision and not an error so we can accept it in its current form and then maybe open a separate issue for discussion within the group? I feel both options can work based on how we want to generate the report.

Reply:
Yeah, I would agree it's more of a design choice. My main concern is not so much that reports should be separately generated for each database but more so that users should be informed about where the hits are derived from if the search is conducted on multiple databases. Let's merge the PR since this is not a pressing issue.